internal/util/netmsg.go

```diff
@@ -27,8 +34,16 @@ func ReadFromUnixSocket(connection net.Conn) ([]byte, error) {
 	return jsonBuffer, nil
 }
 
-func WriteToUnixSocket(connection net.Conn, data []byte) error {
+// Write a message to a network connection.
+// First 8 bytes: the size of the message being written.
+// Remaining bytes: the message content.
+func WriteToNetworkConnection(connection net.Conn, data []byte) error {
 	size := uint64(len(data))
+
+	if size > MAX_FORM_MEM_SIZE_BYTES {
```

Comment:
This is incorrect.

The limit on the size of the message comes from the length of the size value written as the first thing in the connection.
You can make it a constant, but it should not be related to the max form size.
The max form size is a constant right now, but should probably be a config option.
Whereas the max socket message size should just be a large fixed constant.
Just make it something like 2GB and define it here.

Also, defining it as a constant also allows you to reject messages that are too large on the read side as well.